test/cql-pytest/test_guardrail_replication_strategy.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+from util import unique_name, config_value_context
+from cassandra.protocol import ConfigurationException
+
+def create_ks_and_assert_warnings_count(cql, ks_opts, warnings_count = 0, fails_count = 0):
+    keyspace = unique_name()
+    if fails_count:
+        with pytest.raises(ConfigurationException):
+            cql.execute_async("CREATE KEYSPACE " + keyspace + " " + ks_opts).result()
```

Comment:
If this test fails - CREATE KEYSPACE does *not* fail but rather succeeds - it leaves the keyspace behind. Please put inside the "with" also a DROP KEYSPACE, similar to what you did below.
Alternatively you can use the new_test_keyspace() context manager in util.py, which does everything for you (unique name, create keyspace and drop keyspace)

Reply:
Right, done.